sdks/go/pkg/beam/runners/prism/internal/jobservices/management.go

```diff
@@ -188,6 +172,56 @@ func (s *Server) Prepare(ctx context.Context, req *jobpb.PrepareJobRequest) (_ *
 				check("Splittable+Stateful DoFn", "See https://github.com/apache/beam/issues/32139 for information.", "")
 			}
 
+			// Validate whether the triggers on side inputs for are required for
+			// expedient data processing..
+			//
+			// Currently triggered side inputs are not supported by prism, and will
+			// not have early or late firings.
+			//
+			// This feature is required when the Side Input PCollection is unbounded
+			// and is in the a Global Window. This can cause the pipeline to fully
+			// stall while the input is being computed, and may never terminate.
+			//
+			// Other situations may not have desired results, but are valid behaviors
+			// within the model.
+			//
+			// See https://github.com/apache/beam/issues/31438 for implementation tracking.
+			for sideID := range pardo.GetSideInputs() {
+				pcolID := t.GetInputs()[sideID]
+				pcol := job.Pipeline.GetComponents().GetPcollections()[pcolID]
+				wsID := pcol.GetWindowingStrategyId()
+				ws := job.Pipeline.GetComponents().GetWindowingStrategies()[wsID]
+
+				if pcol.GetIsBounded() == pipepb.IsBounded_BOUNDED ||
+					ws.GetWindowFn().GetUrn() != urns.WindowFnGlobal {
+					continue
+				}
+
+				// Within the Unbounded GlobalWindow space is a nich of expressed
+				// user intent that they *do* want to wait for the end of the global
+				// window for output. We should permit these pipelines, as there
+				// is utility for this in testing situations anyway.
+				switch trig := ws.GetTrigger().GetTrigger().(type) {
+				case *pipepb.Trigger_Never_, *pipepb.Trigger_Default_:
+					// Only one firing, at the end of the global window, and is
+					// compatible with Prism's current execution.
+					continue
+				case *pipepb.Trigger_AfterEndOfWindow_:
+					if early := trig.AfterEndOfWindow.GetEarlyFirings(); early == nil || early.GetNever() != nil {
+						if ws.GetAllowedLateness() == 0 {
+							// Late configuration doesn't matter, and there are no early firings.
+							continue
+						}
+						if late := trig.AfterEndOfWindow.GetLateFirings(); late == nil || late.GetNever() != nil {
+							// Lateness allowed, but but no firings anyway.
+							continue
+						}
+					}
+				}
+
+				check("Unbounded GlobalWindow Triggered SideInput", prototext.Format(ws), "See https://github.com/apache/beam/issues/31438 for information.")
```

Comment:
Can we add a little more info like you have above: `Currently triggered side inputs are not supported by prism, and will not have early or late firings.`

Reply:
Also, does `check` actually support a context string like this?